prow/gerrit/client/client.go

```diff
@@ -684,3 +685,21 @@ func (h *gerritInstanceHandler) QueryChangesForProject(log logrus.FieldLogger, p
 		}
 	}
 }
+
+// ChangedFilesProvider lists (in lexicographic order) the files changed as part of a Gerrit patchset.
+// It includes the original paths of renamed files.
+func ChangedFilesProvider(changeInfo *ChangeInfo) config.ChangedFilesProvider {
+	return func() ([]string, error) {
+		changed := sets.NewString()
+		revision := changeInfo.Revisions[changeInfo.CurrentRevision]
```

Comment:
this function takes in pointer, should we check nil pointer here?

Reply:
Will fix in the next PR I'm making. 
/hold cancel